cpp/include/cuspatial/experimental/point_linestring_nearest_points.cuh

```diff
@@ -0,0 +1,103 @@
+ * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam Cart2dItB iterator type for point array of the linestring element of each pair. Must meet
+ * the requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OffsetIteratorA iterator type for `point_geometry_offset` array. Must meet the
+ * requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OffsetIteratorB iterator type for `linestring_geometry_offset` array. Must meet the
+ * requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OffsetIteratorC iterator type for `linestring_part_offset` array. Must meet the
+ * requirements of [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @tparam OutputIt iterator type for output array. Must meet the requirements of
+ * [LegacyRandomAccessIterator][LinkLRAI] and be device-accessible.
+ * @param point_geometry_offset_first beginning of the range of multipoint geometries of each
+ * pair
+ * @param point_geometry_offset_last end of the range of multipoint geometries of each pair
+ * @param points_first beginning of the range of point values
+ * @param points_last end of the range of the point values
+ * @param linestring_geometry_offset_first beginning of the range of offsets to the multilinestring
+ * geometry of each pair, the end range is implied by linestring_geometry_offset_first +
+ * std::distance(`point_geometry_offset_first`, `point_geometry_offset_last`)
+ * @param linestring_offsets_first beginning of the range of offsets to the starting point
+ * of each linestring
+ * @param linestring_offsets_last end of the range of offsets to the starting point
+ * of each linestring
+ * @param linestring_points_first beginning of the range of linestring points
+ * @param linestring_points_last end of the range of linestring points
+ * @param output_first A zipped-iterator of 4 outputs. The first element should be compatible
+ * with iterator_value_type<OffsetIteratorA>, stores the geometry id of the nearest point in
+ * multipoint. The second element should be compatible with iterator_value_type<OffsetIteratorB>,
+ * stores the geometry id of the nearest linestring. The third element should be compatible with
+ * iterator_value_type<OffsetIteratorC>, stores the part id to the nearest segment. The forth
+ * element should be compatible with vec_2d, stores the coordinate of the nearest point on the
+ * (multi)linestring.
+ * @param stream The CUDA stream to use for device memory operations and kernel launches.
+ * @return Output iterator to the element past the last tuple computed.
+ *
+ * @pre all input iterators for coordinates must have `cuspatial::vec_2d` type.
+ * @pre all scalar types must be floating point types, and must be the same type for all input
+ * iterators and output iterators.
```

Comment:
What floating point scalar types are there? Is this precondition necessary?

Reply:
We don't need a line for scalar types. Instead, I think we need a precondition for the base type of the input iterators.

Reply:
>  I think we need a precondition for the base type of the input iterators.

I merged this in the previous precondition.